File: src/main/java/com/sothawo/springdataelastictest/products/Product.java
package com.sothawo.springdataelastictest.products;

import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;
import org.springframework.data.elasticsearch.annotations.ScriptedField;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Document(indexName = "products")
public record Product(
    @Id
    String id,
    @Field(type = FieldType.Text)
    String name,
    @Field(type = FieldType.Double)
    Double netPrice,
    @ScriptedField
    Double grossPrice
) {
}

File: src/main/kotlin/com/sothawo/springdataelastictest/products/Product.kt
package com.sothawo.springdataelastictest.products

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType
import org.springframework.data.elasticsearch.annotations.ScriptedField

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Document(indexName = "products")
data class Product(
    @Id
    val id: String?,
    @Field(type = FieldType.Text)
    val name: String?,
    @Field(type = FieldType.Double)
    val netPrice: Double?,
    @ScriptedField
    val grossPrice: Double?
)
